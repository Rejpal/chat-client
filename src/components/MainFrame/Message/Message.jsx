import React from 'react'
import './Message.css'

export default function Message (props) {
  const { message } = props
  return (
    <div className='Message'>
      <span>{message.author.name}:</span>
      &nbsp;
      <span>{message.content}</span>
    </div>
  )
}
