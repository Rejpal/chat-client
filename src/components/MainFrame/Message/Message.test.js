/* eslint-env mocha, jest */
import Message from './Message'
import { shallow } from 'enzyme'
import React from 'react'
import ReactDOM from 'react-dom'

describe('Message', () => {
  it('shouldRender a message', () => {
    const message = {
      author: { id: '1', name: 'tester' },
      content: 'Test Message'
    }

    expect(shallow(<Message message={message} />)).toMatchSnapshot()
    const div = document.createElement('div')
    ReactDOM.render(<Message message={message} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
