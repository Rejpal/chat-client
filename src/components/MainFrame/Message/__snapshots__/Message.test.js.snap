// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message shouldRender a message 1`] = `
<div
  className="Message"
>
  <span>
    tester
    :
  </span>
  Â 
  <span>
    Test Message
  </span>
</div>
`;
